src/validators/url.rs

```diff
@@ -151,9 +168,10 @@ impl UrlValidator {
     }
 }
 
+// TODO do we still need this?
 enum EitherUrl<'py> {
-    Py(Bound<'py, PyUrl>),
-    Rust(Url),
+    Bound(Bound<'py, PyUrl>),
+    Owned(PyUrl),
```

Comment:
@davidhewitt is there a better way of doing this, or a better name?

Reply:
The point of this is to avoid allocating the Python object until it's absolutely needed. Maybe call it `LazyUrlObject` and have the variants be

```suggestion
    Object(Bound<'py, PyUrl>),
    Struct(PyUrl),
```